interview-questions.md

```diff
@@ -8,89 +8,104 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 5. What is React?
 
   Your answer:
-
+    React is an opensource JavaScript library created by Facebook in 2015. It allows coders to create components that come together to form full user-interfaces. In order to use React within a JavaSctipt file, you need to import React libraries and default files. There is a shortcut to create a new React file using Yarn (also created by Facebook) in your terminal.
+    
   Researched answer:
-
+    React is an opensource JavaScript library created by Facebook in 2015. It allows coders to create components that come together to form full user-interfaces. Compononents are either functional or class-based. React makes app-development very efficient with its use of components and virtual DOM. In order to use React within a JavaSctipt file, you need to import React libraries and default files. There is a shortcut to create a new React file using Yarn (also created by Facebook) in your terminal.
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
   Your answer:
+    Yarn is a technology created by Facebook. It manages and compiles files (like how yarn would stitch things together) together in your compiler. When you run '$ yarn' in a directory, your operating system will download Yarn dependencies so that you can initialize Yarn actiions in that directory. You only need to run '$ yarn' once in your directory to retrieve all the Yarn files and dependencies. Once you run '$ yarn', you can string components together to compile React apps.
 
   Researched answer:
-
+    Yarn is a package manager created by Facebook in order to replace JavaSript Node Package Manager, which is the default method for managing packages in Node.js. Yarn was created to fix the shortcomings of npr. It manages and compiles files (like how yarn would stitch things together) together in your compiler. When you run '$ yarn' in a directory, your operating system will download Yarn dependencies so that you can initialize Yarn actiions in that directory. You only need to run '$ yarn' once in your directory to retrieve all the Yarn files and dependencies. Once you run '$ yarn', you can string components together to compile React apps.
 
 
 7. What is localhost:3000? How do we use localhost:3000 in the development process?
 
   Your answer:
+    localhost:3000 is your local server, or your operating system. We use localhost:3000 to "locally host", which means that it runs your file on a web browser directly from your computer. This is important in the development process because you want to be able to test your code as much as possible before needing to finally push it onto a public domain. When you locally host, you can look at the console to debug portions of your code and you can also preview what the user-interface will look like.
```

Comment:
Totally. I think it would also be fair to say that 'localhost' is a bit of a placeholder for your specific machine's IP address, while the :3000 is the port in which it accesses it.